modules/standard/DynamicIters.chpl

```diff
@@ -154,6 +154,113 @@ where tag == iterKind.follower
   }
 }
 
+//************************* Multidimensional Dynamic iterator
+/*
+
+  :arg c: The domain to iterate over. The rank of the domain must be greater 
+          than one. If using a 1d domain, the user should use the existing
+          dynamic iterators (use DynamicIters). 
```

Comment:
Is this a technical limitation? If not, I think we would want to support 1D domains for completeness. We still find users writing code like this at times:

```chpl
forall i dynamic({1..10}, 1) { ... }
```

instead of passing in a range directly. It's not the style we suggest, but if we can handle it I think that's better than a compile-time error.